lib/errors.js

```diff
@@ -1,17 +1,56 @@
-'use strict';
+const statuses = require('statuses');
 
-const util = require('util');
-
-function CustomError (message) {
-  Error.captureStackTrace(this, this.constructor);
-  this.name = this.constructor.name;
-  this.message = message;
+class CustomError extends Error {
+  constructor (message) {
+    super();
+    this.message = message;
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
+  }
 }
-util.inherits(CustomError, Error);
 
 class ConfigurationError extends CustomError {}
 
+class GatewayError extends CustomError {
```

Comment:
Why do we need this custom Error ?

Reply:
Idk, it was here before.

Reply:
In my opinion it should be fixed in #63, #185